data-prepper-api/src/main/java/org/opensearch/dataprepper/model/metric/JacksonMetric.java

```diff
@@ -32,13 +32,17 @@ public abstract class JacksonMetric extends JacksonEvent implements Metric {
     protected static final String SCHEMA_URL_KEY = "schemaUrl";
     protected static final String EXEMPLARS_KEY = "exemplars";
     protected static final String FLAGS_KEY = "flags";
+    private boolean flattenAttributes = true;
```

Comment:
I don't think changing the state in this class is the right approach here.

Reply:
I agree that we need a flag to change otel metrics to NOT flatten the attributes. But we do not think metrics generated in the data prepper to support both modes would be the correct thing. To be compliant with OTEL format, we should remove the flattening of the attributes. Keeping it to just so that it will have backward compatibility with the existing customer use cases.

Reply:
Also, each instance of otel metrics is different from each other, do we need to put the entire OTEL metrics in one more or the other? Why not provide instance level flexibility to be in one more or the other? I can move the flag to constructor, if that makes it a cleaner solution.